### ISSUE TITLE
[🐞]: Missing Recipes 1.21.1 and a crash

### ISSUE THREAD

**Issue by Sinogard** [MehVahdJukaar/WoodGood#657]
### Before Continuing:

- [X] Always check you are using the latest version of the mods and its dependencies
- [X] Remove mod that enhances Minecraft: Optifine, Sodium, others. The issue still persists.
- [X] If you are unsure which mod is the culprit.
Disable all of your mods and enable them 1-2 mods each time to isolate the culprit
- [X] Confirm that there is no existing issue with a similar description submitted in the list of issues.

### Version

1.21

### Loader

Fabric

### EveryCompat Version

1.21-2.72

### Moonlight Lib Version

1.21-2.15.2

### Issue with mods

Betterend 21.0.11
Betternether 21.0.11
Wilder Wilds 3.0.4-mc1.21.1
chipped 4.0.2
Macaw's Furniture 3.3.0, Fences 1.1.2, Lights and Lamps 1.1.1, Paths and Pavings, 1.0.5
Handcrafted 4.0.2

### Issue Detail

Recipes for each of these are not generated between each other. the textures and blocks are. they also fail to show in the creative mode tabs. they do show in EMI. as I tested using EMI to Acquire the generated blocks i crashed when placing Macaws Furniture blocks generated by Every Compat for Wilder Wilds (Barbola Counter).

### OPTIONAL: Latest.log | Crash-report Attachment

[latest.log](https://github.com/user-attachments/files/17598089/latest.log)
Crash Log
[latest.log](https://github.com/user-attachments/files/17598148/latest.log)

**Comment by Xelbayria** [comment]
> i crashed when placing Macaws Furniture blocks generated by Every Compat for Wilder Wilds (Barbola Counter).

both `latest.log` seems to be similar. I need the crash-report. I can find out where is the problem

>  they also fail to show in the creative mode tabs.

I know about the tab problem. 

> Recipes for each of these are not generated between each other

I'm aware of this problem, too. Cannot say when will it be fixed.

**Comment by Sinogard** [comment]
This is where it crashes. Minecraft basically freezes and the server shuts down. it stays open but its just in a perpetual hang.


<details>
<summary>Error Log</summary>

```
[11:54:48] [Render thread/ERROR]: Unreported exception thrown!
java.lang.IllegalStateException: Invalid block entity mcwfurnitures:box_block // net.kikoz.mcwfurnitures.storage.FurniturekEntity state at class_2338{x=-658, y=60, z=1525}, got Block{everycomp:mcfur/wilderwild/stripped_baobab_cupboard_counter}[facing=north,hinge=left]
at knot/net.minecraft.class_2586.a(class_2586.java:57) ~[client-intermediary.jar:?]
at knot/net.minecraft.class_2586.<init>(class_2586.java:51) ~[client-intermediary.jar:?]
at knot/net.kikoz.mcwfurnitures.storage.FurniturekEntity.<init>(FurniturekEntity.java:22) ~[mcw-furniture-3.3.0-mc1.21fabric.jar:?]
at knot/net.kikoz.mcwfurnitures.objects.FurnitureObjectNonFaceable.method_10123(FurnitureObjectNonFaceable.java:82) ~[mcw-furniture-3.3.0-mc1.21fabric.jar:?]
at knot/net.minecraft.class_2818.method_12010(class_2818.java:286) ~[client-intermediary.jar:?]
at knot/net.minecraft.class_1937.method_30092(class_1937.java:232) ~[client-intermediary.jar:?]
at knot/net.minecraft.class_638.method_30092(class_638.java:160) ~[client-intermediary.jar:?]
at knot/net.minecraft.class_1937.method_8652(class_1937.java:217) ~[client-intermediary.jar:?]
at knot/net.minecraft.class_1747.method_7708(class_1747.java:153) ~[client-intermediary.jar:?]
at knot/net.minecraft.class_1747.method_7712(class_1747.java:76) ~[client-intermediary.jar:?]
at knot/net.minecraft.class_1747.method_7884(class_1747.java:46) ~[client-intermediary.jar:?]
at knot/net.minecraft.class_1799.method_7981(class_1799.java:373) ~[client-intermediary.jar:?]
at knot/net.minecraft.class_636.method_41934(class_636.java:363) ~[client-intermediary.jar:?]
at knot/net.minecraft.class_636.method_41933(class_636.java:318) ~[client-intermediary.jar:?]
at knot/net.minecraft.class_636.method_41931(class_636.java:279) ~[client-intermediary.jar:?]
at knot/net.minecraft.class_636.method_2896(class_636.java:317) ~[client-intermediary.jar:?]
at knot/net.minecraft.class_310.method_1583(class_310.java:1852) ~[client-intermediary.jar:?]
at knot/net.minecraft.class_310.redirect$bhf000$betterthirdperson$onDoItemUse(class_310.java:8734) ~[client-intermediary.jar:?]
at knot/net.minecraft.class_310.method_1508(class_310.java:2111) ~[client-intermediary.jar:?]
at knot/net.minecraft.class_310.method_1574(class_310.java:1938) ~[client-intermediary.jar:?]
at knot/net.minecraft.class_310.method_1523(class_310.java:1246) ~[client-intermediary.jar:?]
at knot/net.minecraft.class_310.method_1514(class_310.java:882) [client-intermediary.jar:?]
at knot/net.minecraft.client.main.Main.main(Main.java:256) [client-intermediary.jar:?]
at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480) [fabric-loader-0.16.9.jar:?]
at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.16.9.jar:?]
at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.16.9.jar:?]
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?]
at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[?:?]
at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:243) [NewLaunch.jar:?]
at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:278) [NewLaunch.jar:?]
at org.multimc.EntryPoint.listen(EntryPoint.java:143) [NewLaunch.jar:?]
at org.multimc.EntryPoint.main(EntryPoint.java:34) [NewLaunch.jar:?]
[11:54:50] [Server thread/INFO]: Stopping server
[11:54:50] [Server thread/INFO]: Saving players
[11:54:50] [Server thread/INFO]: SinogardNunitsuj lost connection: Disconnected
[11:54:50] [Server thread/INFO]: SinogardNunitsuj left the game
[11:54:50] [Server thread/INFO]: Updating all forceload tickets for 3764f41d-24b2-4e8d-aca5-e688821587c3
[11:54:50] [Server thread/INFO]: Stopping singleplayer server as player logged out
[11:54:50] [Server thread/INFO]: Saving worlds
[11:54:50] [Server thread/INFO]: ServerLevel[Sincraft S2 Survival]/minecraft:overworld: waiting for 8410 chunks to unload
[11:54:51] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Sincraft S2 Survival]'/minecraft:overworld
[11:54:51] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Sincraft S2 Survival]'/minecraft:the_end
[11:54:51] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Sincraft S2 Survival]'/minecraft:the_nether
[11:54:51] [Server thread/INFO]: ThreadedAnvilChunkStorage (Sincraft S2 Survival): All chunks are saved
[11:54:51] [Server thread/INFO]: ThreadedAnvilChunkStorage (DIM1): All chunks are saved
[11:54:51] [Server thread/INFO]: ThreadedAnvilChunkStorage (DIM-1): All chunks are saved
[11:54:51] [Server thread/INFO]: ThreadedAnvilChunkStorage: All dimensions are saved
[11:55:00] [MicrophoneThread/INFO]: [voicechat] Connection timeout
[11:55:00] [MicrophoneThread/INFO]: [voicechat] Stopping microphone thread
[11:55:00] [MicrophoneThread/INFO]: [voicechat] Disconnecting voicechat

```

</details>

**Comment by Xelbayria** [comment]
I know why.

**Comment by Xelbayria** [comment]
> placing Macaws Furniture blocks generated by Every Compat for Wilder Wilds (Barbola Counter).

### ☝🏻 FIXED in v2.7.3

**Comment by DarthCaesium** [comment]
Can confirm this also happens on NeoForge.

**Comment by Xelbayria** [comment]
> Can confirm this also happens on NeoForge.

We are aware of this. The issue is in Moonlight. it's quite tricky and difficult to pinpoint the problem and fix it. Of course, I'm talking about the recipe. 

it's currently WIP.

**Comment by Xelbayria** [comment]
Good news. 

## The recipe issue is fixed in **Moonlight Lib v2.15.10**
